### ISSUE TITLE
Training falled after 60000 iterations

### ISSUE THREAD

**Issue by qiuqiangkong** [soroushmehr/sampleRNN_ICLR2017#26]
Hi, really brilliant code! when I run the pytorch implementation, I found the trianing and validation loss increases dramtically after 60000 iterations. The loss curve looks like: https://drive.google.com/open?id=1_64-jD3hOtXmrOoVMq5hD8pWfmEUc7yE

Do you have any idea of this increased loss? Thank you very much! 

Qiuqiang

**Comment by Cortexelus** [comment]
The pytorch implementation doesn’t behave exactly the same as theano
version. For example, they implement normalization differently. And even if
they were coded the same, frameworks make different-level assumptions, and
it doesnt translate perfectly. My pytorch version runs faster but the
quality isnt as good. It’s notoriously hard to debug moving a model from
one framework to another.

One thing the pytorch version could benefit from is adding skip
connections. I recommend doing it and submitting a PR :)

As for why gradient descent does that sometimes, good question. I don’t
know how to debug that. Maybe it got stuck in saddle and finally freed
itself? No idea. Maybe try a different optimizer or learning rate. I saw a
good poster at ICLR about cyclic learning rates.

**Comment by qiuqiangkong** [comment]
Many thanks for the reply! I fixed out the problem. In the adam optimizer, I need to set this flag to true: amsgrad=True. Then it will fix this problem.

**Comment by guozixunnicolas** [comment]
@qiuqiangkong Hi qiuqiang, I am using tensorflow to implement SampleRNN and Ive encountered the same problem, however I dont think in tf.train.adamoptimizer they have this amsgrad flag. Any idea how to troubleshoot this? Thanks in advance!

**Comment by guozixunnicolas** [comment]
![image](https://user-images.githubusercontent.com/44756093/65658936-48880e80-e05c-11e9-9b27-ffd0e2745270.png)
something like this

**Comment by qiuqiangkong** [comment]
@zguo008 Interesting! I think keras has the amsgrad flag. Or try Rmsprop optimizer instead. I guess this phenomenon is caused by optimization. Let me know if you still not solve this problem.

**Comment by guozixunnicolas** [comment]
Hi qiuqiang, thanks for your reply:) I’ll try RMSprop later. I tuned down value of epsilon in Adam optimizer and it seems that there’s no sudden increase in loss for now. Thank you for ur answer! It’s some great help

**Comment by qiuqiangkong** [comment]
@zguo008 That is great! What epsilon are you using now?
